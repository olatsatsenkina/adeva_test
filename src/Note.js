import React from 'react';

export default class Note extends React.Component {

  handleClick(e) {
    let {index, handler} = this.props;
    handler(index, 'edit');
  }

  render() {
    let {text} = this.props;
    text = text.replace(/&#44;/g, ",");
    return (
        <li onClick={this.handleClick.bind(this)}>{text}</li>
    );
  }
}
