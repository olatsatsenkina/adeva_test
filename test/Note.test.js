import React from 'react';
import Note from '../src/Note.js';
import renderer from 'react-test-renderer';

it('Note renders correctly', () => {
  const tree = renderer
    .create(<Note
      text = "Test Note"
      index="0"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

let handler = (index, mode) => {
  expect(index).toBe("0");
  expect(mode).toBe("edit");
}

test('OnClick handler is working', () => {
  const noteComponent = renderer.create(<Note text="Test" handler={handler} index="0"/>);
  const rootInstance = noteComponent.root;
  const li = rootInstance.findByType("li");
  li.props.onClick();
});
