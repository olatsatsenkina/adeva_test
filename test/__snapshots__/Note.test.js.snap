// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders correctly 1`] = `
<li
  onClick={[Function]}
>
  Test Note
</li>
`;
